tools/checkpoint_conversion/convert_gpt_neox_checkpoints.py

```diff
@@ -0,0 +1,179 @@
+
+keras_model.get_layer("token_embedding").embeddings.assign(
+    hf_model.embed_in.weight.detach().numpy()
+)
+
+keras_model.get_layer("token_embedding").embeddings.assign(
+    hf_model.embed_in.weight.detach().numpy()
+)
+
+for ilayer in range(cfg["num_layers"]):
+    # attention layer
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._qkv_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.attention.query_key_value.weight"]
+        .numpy()
+        .T.reshape((cfg["hidden_dim"], cfg["num_heads"], -1))
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._qkv_dense.bias.assign(
+        hf_wts[f"layers.{ilayer}.attention.query_key_value.bias"].reshape(
+            (cfg["num_heads"], -1)
+        )
+    )
+
+    # Attention Dense
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._output_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.attention.dense.weight"].numpy().T
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layer._output_dense.bias.assign(
+        hf_wts[f"layers.{ilayer}.attention.dense.bias"]
+    )
+
+    # LAYERNORM
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layernorm.gamma.assign(
+        hf_wts[f"layers.{ilayer}.input_layernorm.weight"]
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._self_attention_layernorm.beta.assign(
+        hf_wts[f"layers.{ilayer}.input_layernorm.bias"]
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_layernorm.gamma.assign(
+        hf_wts[f"layers.{ilayer}.post_attention_layernorm.weight"]
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_layernorm.beta.assign(
+        hf_wts[f"layers.{ilayer}.post_attention_layernorm.bias"]
+    )
+
+    # MLP
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_intermediate_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.mlp.dense_h_to_4h.weight"].numpy().T
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_intermediate_dense.bias.assign(
+        hf_wts[f"layers.{ilayer}.mlp.dense_h_to_4h.bias"]
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_output_dense.kernel.assign(
+        hf_wts[f"layers.{ilayer}.mlp.dense_4h_to_h.weight"].numpy().T
+    )
+
+    keras_model.get_layer(
+        f"transformer_layer_{ilayer}"
+    )._feedforward_output_dense.bias.assign(
+        hf_wts[f"layers.{ilayer}.mlp.dense_4h_to_h.bias"]
+    )
+
+
+keras_model.get_layer("layer_norm").gamma.assign(
+    hf_wts["final_layer_norm.weight"]
+)
+
+keras_model.get_layer("layer_norm").beta.assign(hf_wts["final_layer_norm.bias"])
+
+hf_tokenizer = AutoTokenizer.from_pretrained(PRESET)
```

Comment:
we should update this section to check tokenizer output for some simple input as well, now that we have added the tokenizer here.

Reply:
Added this part, not using tokenizer input for model as we still don't have preprocessor.
outputs of tokenizer are same as hf tokenizer.
By the way they are using `gpt-neox-20b` vocabulary for pythia suite.